build-source-tarball.sh

```diff
@@ -87,6 +87,10 @@ git submodule foreach --quiet --recursive '
         git --work-tree="$TARBALL_SUBMODULE_PATH" checkout -- .
     fi'
 
+echo 'Removing binaries from tarball src...'
+find $TARBALL_ROOT/src -type f -iname *.dll -exec rm {} \;
+find $TARBALL_ROOT/src -type f -iname *.exe -exec rm {} \;
```

Comment:
You can combine the lines with `-o` like this:

https://github.com/dotnet/source-build/blob/84be767a694f2bb6a145ebf8090e03c798ac0eb2/.vsts.pipelines/jobs/ci-linux.yml#L147-L151